test/PlaneWaveBasis.jl

```diff
@@ -112,20 +112,21 @@ end
     @test length(G_vectors(fft_size)) == prod(fft_size)
     @test length(r_vectors(basis))    == prod(fft_size)
 
-    @test all(G_vectors(basis)      .== G_vectors(fft_size))
-    @test all(G_vectors_cart(basis) .== map(G -> model.recip_lattice * G,
+    # Using `isapprox` due to https://github.com/JuliaMolSim/DFTK.jl/pull/727#issuecomment-1246749288
+    @test all(G_vectors(basis)      .≈ G_vectors(fft_size))
+    @test all(G_vectors_cart(basis) .≈ map(G -> model.recip_lattice * G,
```

Comment:
Hm, that's very weird. Inlining or not should not change semantics, even in floating point. It may end up being a julia bug... Can you maybe reduce it to a self-contained example (without DFT)?

Reply:
Fastmath might potentially change things, but we don't use fastmath...